src/libraries/NormalizedAmount.sol

```diff
@@ -1,58 +1,172 @@
 // SPDX-License-Identifier: Apache 2
+/// @dev NormalizedAmount is a utility library to handle token amounts with different decimals
 pragma solidity >=0.8.8 <0.9.0;
 
-type NormalizedAmount is uint64;
-
-using {gt as >, lt as <, sub as -, add as +, div, min, unwrap} for NormalizedAmount global;
-
-function gt(NormalizedAmount a, NormalizedAmount b) pure returns (bool) {
-    return NormalizedAmount.unwrap(a) > NormalizedAmount.unwrap(b);
+struct NormalizedAmount {
+    uint64 amount;
+    uint8 decimals;
 }
 
-function lt(NormalizedAmount a, NormalizedAmount b) pure returns (bool) {
-    return NormalizedAmount.unwrap(a) < NormalizedAmount.unwrap(b);
+function minUint8(uint8 a, uint8 b) pure returns (uint8) {
+    return a < b ? a : b;
 }
```

Comment:
very nitpicky but what do you think of moving this method to `EndpointHelpers.sol`
There is a `min(uint256,uint256)` method already in there.